Neos.Fusion/Classes/Core/FusionSourceCode.php

```diff
@@ -0,0 +1,61 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Fusion\Core;
+
+/*
+ * This file is part of the Neos.Fusion package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+
+#[Flow\Proxy(false)]
+class FusionSourceCode
+{
+    private function __construct(
+        private string $sourceCode,
+        private ?string $contextPathAndFilename
+    ) {
+        if (trim($sourceCode) === '') {
+            throw FusionSourceCodeIsInvalid::becauseTheSourceCodeIsEmpty($this->contextPathAndFilename);
+        }
+    }
+
+    public static function fromString(string $string): self
+    {
+        return new self($string, null);
+    }
+
+    public static function fromFile(string $fileName): self
+    {
+        $sourceCode = file_get_contents($fileName);
```

Comment:
should this `file_get_contents` happen lazily?

eg - it might happen that the `FusionSourceCode` is already cached and then we would not need its contents